### PR TITLE
Desul atomics: fix bug in `desul::Impl::numeric_limits_max<uint64_t>` value

### PR DESCRIPTION
**Pull request by dalg24**
Use modulo arithmetic instead of wrong hex number

### CODE REVIEW

File: tpls/desul/include/desul/atomics/Common.hpp [hunk 1]
```diff
@@ -83,11 +83,11 @@ struct numeric_limits_max;
 
 template <>
 struct numeric_limits_max<uint32_t> {
-  static constexpr uint32_t value = 0xffffffffu;
+  static constexpr uint32_t value = -1;
 };
 template <>
 struct numeric_limits_max<uint64_t> {
-  static constexpr uint64_t value = 0xfffffffflu;
```
Comment (dalg24) [inline tpls/desul/include/desul/atomics/Common.hpp:90]: The right value would have been `0xffffffffffffffff`.  The suffixes are pointless here.

File: tpls/desul/include/desul/atomics/Common.hpp [hunk 2]
```diff
@@ -83,11 +83,11 @@ struct numeric_limits_max;
 
 template <>
 struct numeric_limits_max<uint32_t> {
-  static constexpr uint32_t value = 0xffffffffu;
+  static constexpr uint32_t value = -1;
```
Comment (PhilMiller) [inline tpls/desul/include/desul/atomics/Common.hpp:86]: An alternative expression that makes the 'all ones' nature totally clear, and avoids pedantic details about types and numeric behavior, would be
```suggestion
  static constexpr uint32_t value = ~static_cast<uint32_t>(0);
```

Comment (crtrott) [reply tpls/desul/include/desul/atomics/Common.hpp:86]: I like this.

Comment (crtrott) [reply tpls/desul/include/desul/atomics/Common.hpp:86]: This being Phil's suggestion.

Comment (dalg24) [reply tpls/desul/include/desul/atomics/Common.hpp:86]: Discussion needs to happen on the Desul side and until proof of the contrary no warnings are being emitted with the code I proposed and modulo arithmetic is as easy as it gets so there is no need to try to be clever.

**Review by crtrott — APPROVED** [review]

File: tpls/desul/include/desul/atomics/Common.hpp (hunk 2 above)
Comment (crtrott) [reply tpls/desul/include/desul/atomics/Common.hpp:86]: yeah its fine leave the -1

**Review by Rombur — APPROVED** [review]

**Pull request merged**